models/iosxr/interfaces/iosxr_interface.yaml

```diff
@@ -0,0 +1,93 @@
+Igenerator_version: 1.0
+metadata:
+  version: 1.1
+  status:
+  - preview
+  supported_by:
+  - 'Ansible Network'
+  copyright_str: Copyright 2019 Red Hat
+  license: gpl-3.0.txt
+info:
+  network_os: iosxr
+  resource: interfaces
+  version_added: 2.9
+  short_description: 'Manage Interface on Cisco IOS network devices.'
+  description:
+  - Manage Interface on Cisco IOS-XR network devices.
+  authors:
+  - Sumit Jaiswal (@justjais)
+  extends_documentation_fragment: iosxr
+  notes:
+    - Tested against IOS-XR Version 6.1.3 on VIRL
+schema:
+  type: object
+  description: The schema used for the argspec and docstring
+  properties:
+    config:
+      type: array
+      description: The provided configuration
+      items:
+        $ref: '#/definitions/config_dict'
+    state:
+      $ref: '#/definitions/state'
+
+definitions:
+  config_dict:
+    description:
+    - The some_dict.
+    type: object
+    properties:
+      name:
+        description:
+        - Name of the interface to configure in C(type + path) format. e.g. C(GigabitEthernet0/0/0/0)
+        type: str
+        required: true
+      description:
+        description:
+        - Description of Interface being configured.
+        type: str
+      enabled:
+        description:
+        - Removes the shutdown configuration, which removes the forced administrative down on the interface,
+          enabling it to move to an up or down state.
+        type: bool
+        default: true
+      active:
+        description:
+        - Whether the interface is C(active) or C(preconfigured). Preconfiguration allows you to configure modular
+          services cards before they are inserted into the router. When the cards are inserted, they are instantly
+          configured. Active cards are the ones already inserted.
+        type: str
+        choices: ['active', 'preconfigure']
+        default: active
+      speed:
+        description:
+        - Configure the speed for an interface. Default is auto-negotiation when not configured.
+        type: str
+        choices: ['10', '100', '1000']
+      mtu:
+        description:
+        - Sets the MTU value for the interface. Range is between 64 and 65535. Applicable for Ethernet interface only.
+        type: str
+      duplex:
+        description:
+        - Configures the interface duplex mode. Default is auto-negotiation when not configured.
+        type: str
+        default: auto
+        choices: ['full', 'half']
```

Comment:
other resource models include 'auto' in the list of choices here...

Reply:
In `IOSXR` supported duplex options are `full` and `half`.